resotocore/resotocore/config/config_override_service.py

```diff
@@ -0,0 +1,133 @@
+from typing import List, Optional, Callable, Dict, Any, Awaitable
+from pathlib import Path
+import yaml
+import logging
+import asyncio
+from resotocore.types import Json
+from resotocore.ids import ConfigId
+from resotocore.util import deep_merge
+from resotocore.config import ConfigOverride
+from deepdiff import DeepDiff
+import aiofiles.os as aos
+
+log = logging.getLogger("config_override_service")
+
+
+class ConfigOverrideService(ConfigOverride):
+    def __init__(self, override_paths: List[Path], sleep_time: float = 3.0):
+        self.override_paths = override_paths
+        self.sleep_time = sleep_time
+        self.overrides: Dict[ConfigId, Json] = {}
+        self.stop_watcher = asyncio.Event()
+        self.override_change_hooks: List[Callable[[Dict[ConfigId, Json]], Awaitable[Any]]] = []
+        self.watcher_task: Optional[asyncio.Task[Any]] = None
+        self.overrides = self._load_overrides()
+        self.mtime_hash: int = 0
+
+    def add_override_change_hook(self, hook: Callable[[Dict[ConfigId, Json]], Awaitable[Any]]) -> None:
+        self.override_change_hooks.append(hook)
+
+    # no async here because page cache will keep this in memory anyway
+    def _load_overrides(self, silent: bool = False) -> Dict[ConfigId, Json]:
+        if not self.override_paths:
+            return {}
+
+        # all config files that will be used
+        config_files: List[Path] = []
+        # collect them all
+        for path in self.override_paths:
+            if path.is_dir():
+                config_files.extend(
+                    [file for file in path.iterdir() if file.is_file() and file.suffix in (".yml", ".yaml")]
+                )
+            else:
+                config_files.append(path)
+
+        # json with all merged overrides for all components such as resotocore, resotoworker, etc.
+        overrides_json: Json = {}
+        # merge all provided overrides into a single object, preferring the values from the last override
+        for config_file in config_files:
+            with config_file.open() as f:
```

Comment:
Consider using aiofiles for reading the files.

Reply:
I wanted to keep it like that for the reason of getting a fully constructed object after calling init.